src/nvim/lua/converter.c

```diff
@@ -766,8 +766,6 @@ void nlua_push_Object(lua_State *lstate, Object *obj, bool special)
     break;
   case kObjectTypeLuaRef: {
     nlua_pushref(lstate, obj->data.luaref);
-    api_free_luaref(obj->data.luaref);
-    obj->data.luaref = LUA_NOREF;
```

Comment:
hmm, this would cause leaks in cases a arena allocated value is returned with included luarefs which are unique. 

But perhaps autocmd was the only case where we did this? I thought it was more but I could remember wrong ( speaking about real useful functions, not the `nvim__id` clownshow).

Reply:
If I understand the meaning of "unique luarefs" correctly, after this PR an arena allocated value shouldn't include unique luarefs, because `api_nvim_luaref()` is no longer used in an arena-allocated value.

Reply:
Ye I understand the intention. As the tests don't catch any new leaks, it is probably correct.